src/HypothesisClient/ApiClient/TokenClient.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace eLife\HypothesisClient\ApiClient;
+
+use eLife\HypothesisClient\Credentials\JWTSigningCredentials;
+use eLife\HypothesisClient\HttpClient\HttpClient;
+use eLife\HypothesisClient\HttpClient\UserAgentPrependingHttpClient;
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Psr7\Uri;
+
+final class TokenClient
+{
+    use ApiClient;
+
+    private $credentials;
+
+    public function __construct(HttpClient $httpClient, JWTSigningCredentials $credentials, array $headers = [])
+    {
+        $this->httpClient = new UserAgentPrependingHttpClient($httpClient, 'HypothesisClient');
```

Comment:
I think we usually prepend the name of the project in the `User-Agent` too 
e.g. `eLifeJournal/dev-master@7b88378 eLifeApiSdk/dev-master@af6b160 eLifeApiClient/dev-master@dba0869 GuzzleHttp/6.2.1 curl/7.35.0 PHP/7.0.23-1+ubuntu14.04.1+deb.sury.org+1`
which here would only need `eLifeAnnotations/dev-master` as the second parameter (let's ignore the commit for now)

Reply:
Perhaps this should be set in configuration with `HypothesisClient` as the default?

Reply:
Should not be able to change by environment so I think not a good candidate for configuration. Will need revising anyway when/if we extract `eLife\HypothesisClient` as a library from here